animatch/CommonViews/NGGSecureField.swift

```diff
@@ -0,0 +1,73 @@
+//
+//  NGGSecureField.swift
+//  animatch
+//
+//  Created by Ксения Панкратова on 23.11.2024.
+//
+
+import SwiftUI
+
+struct NGGSecureField: View {
+    @Binding private var text: String
+    @State private var isSecured: Bool = true
+    private var title: String
+    
+    init(_ title: String, text: Binding<String>) {
+        self.title = title
+        self._text = text
+    }
+    
+    var body: some View {
+        ZStack(alignment: .trailing) {
+            Group {
+                if isSecured {
+                    StyledSecureField(title: title, text: $text)
+                    
+                } else {
+                    NGGTextField(title: title, text: $text)
+                }
+            }
+            
+            Button(action: {
+                isSecured.toggle()
+            }) {
+                Image(systemName: self.isSecured ? "eye.slash" : "eye")
+                    .accentColor(.white)
+                    .opacity(0.8)
+            }
+            .padding(.bottom, 30)
+            .padding(.trailing, 5)
+        }
+    }
+}
+
+private struct StyledSecureField: View {
```

Comment:
Надо добавить Марку. Всегда разделяй через MARK если новые структуры или что-то такое. это более читаемо

```suggestion

// MARK: - StyledSecureField

private struct StyledSecureField: View {
```